src/modules/network/implementation/libp2p-service.js

```diff
@@ -214,38 +212,65 @@ class Libp2pService {
+                } catch (error) {
+                    this.logger.warn(
+                        `Request message is not valid with operationId:${message.header.operationId}, peerId:${peerIdString}, keyword:${message.header.keywordUuid}, Error:${error}`,
+                    );
+                }
+
                 this.removeCachedSession(
                     message.header.operationId,
                     message.header.keywordUuid,
                     peerIdString,
                 );
             } else if (busy) {
-                await this.sendMessageResponse(
-                    protocol,
-                    peerIdString,
-                    NETWORK_MESSAGE_TYPES.RESPONSES.BUSY,
-                    message.header.operationId,
-                    message.header.keywordUuid,
-                    {},
-                );
+                try {
+                    await this.sendMessageResponse(
+                        protocol,
+                        peerIdString,
+                        NETWORK_MESSAGE_TYPES.RESPONSES.BUSY,
+                        message.header.operationId,
+                        message.header.keywordUuid,
+                        {},
+                    );
+                } catch (error) {
+                    this.logger.warn(
+                        `Peer is busy, operationId:${message.header.operationId}, peerId:${peerIdString}, keyword:${message.header.keywordUuid}, Error: ${error}`,
+                    );
+                }
                 this.removeCachedSession(
                     message.header.operationId,
                     message.header.keywordUuid,
                     peerIdString,
                 );
             } else {
                 this.logger.debug(
-                    `Receiving message from ${peerIdString} to ${this.config.id}: protocol: ${protocol}, messageType: ${message.header.messageType};`,
+                    `Receiving message from ${peerIdString} to ${this.config.id}: protocol: ${protocol} with operationId:${message.header.operationId} and keyword:${message.header.keywordUuid}, messageType: ${message.header.messageType};`,
                 );
-                await handler(message, peerIdString);
+                try {
+                    await handler(message, peerIdString);
+                } catch (error) {
+                    await this.sendMessageResponse(
+                        protocol,
+                        peerIdString,
+                        NETWORK_MESSAGE_TYPES.RESPONSES.NACK,
+                        message.header.operationId,
+                        message.header.keywordUuid,
+                        { errorMessage: 'Unable to handle request' },
+                    );
+                    this.logger.error(
+                        `Error handling message from ${peerIdString} to ${this.config.id}: protocol :${protocol} with operationId${message.header.operationId} and keyword:${message.header.keywordUuid}, Error : ${error} `,
```

Comment:
Please update log messages to be consistent everywhere in PR, they should have following format `word: ${value}`, so some word, semi-colons followed by space and after that a value